src/modules/networks/Networks.ts

```diff
@@ -76,10 +76,22 @@ export class Networks {
   private ethereumChainConfigs: EthereumChainConfig[];
   private visitedChains: Set<string>;
 
-  static getChainId(network: NetworkConfig) {
-    return network.standard === 'eip155'
-      ? normalizeChainId(network.specification.eip155.id)
-      : null;
+  static getChainId(
+    network: Partial<
+      Pick<NetworkConfig, 'standard' | 'specification' | 'external_id'>
+    >
+  ) {
+    if ('standard' in network && network.specification) {
+      return network.standard === 'eip155'
+        ? normalizeChainId(network.specification.eip155.id)
+        : null;
+    } else if (network.external_id) {
```

Comment:
Why do we need this condition? external_id is deprecated and shouldn't be persisted anywhere any more.

Reply:
This method is called by code that deals with the persisted network configs